drivers/sensor/bosch/bmp388/bmp388.c

```diff
@@ -577,26 +575,27 @@ static int bmp388_init(const struct device *dev)
 #define BMP388_INT_CFG(inst)
 #endif
 
-#define BMP388_INST(inst)						   \
-	static struct bmp388_data bmp388_data_##inst = {		   \
-		.odr = DT_INST_ENUM_IDX(inst, odr),			   \
-		.osr_pressure = DT_INST_ENUM_IDX(inst, osr_press),	   \
-		.osr_temp = DT_INST_ENUM_IDX(inst, osr_temp),		   \
-	};								   \
-	static const struct bmp388_config bmp388_config_##inst = {	   \
-		BMP388_BUS_CFG(inst)					   \
-		BMP388_INT_CFG(inst)					   \
-		.iir_filter = DT_INST_ENUM_IDX(inst, iir_filter),	   \
-	};								   \
-	PM_DEVICE_DT_INST_DEFINE(inst, bmp388_pm_action);		   \
-	SENSOR_DEVICE_DT_INST_DEFINE(					   \
-		inst,							   \
-		bmp388_init,						   \
-		PM_DEVICE_DT_INST_GET(inst),				   \
-		&bmp388_data_##inst,					   \
-		&bmp388_config_##inst,					   \
-		POST_KERNEL,						   \
-		CONFIG_SENSOR_INIT_PRIORITY,				   \
-		&bmp388_api);
-
-DT_INST_FOREACH_STATUS_OKAY(BMP388_INST)
+#define BMP3XX_INST(inst, chipid)\
+	static struct bmp388_data bmp388_data_##inst##chipid = {\
+		.odr = DT_INST_ENUM_IDX(inst, odr),\
+		.osr_pressure = DT_INST_ENUM_IDX(inst, osr_press),\
+		.osr_temp = DT_INST_ENUM_IDX(inst, osr_temp),\
+		.chip_id = chipid,\
+	};\
+	static const struct bmp388_config bmp388_config_##inst##chipid = {\
+		BMP388_BUS_CFG(inst)\
+		BMP388_INT_CFG(inst)\
+		.iir_filter = DT_INST_ENUM_IDX(inst, iir_filter),\
+	};\
+	PM_DEVICE_DT_INST_DEFINE(inst, bmp388_pm_action);\
+	SENSOR_DEVICE_DT_INST_DEFINE(inst, bmp388_init, PM_DEVICE_DT_INST_GET(inst),\
+				    &bmp388_data_##inst##chipid, &bmp388_config_##inst##chipid,\
+					POST_KERNEL, CONFIG_SENSOR_INIT_PRIORITY, &bmp388_api);
+
+#define DT_DRV_COMPAT  bosch_bmp388
+DT_INST_FOREACH_STATUS_OKAY_VARGS(BMP3XX_INST, BMP388_ID)
+#undef DT_DRV_COMPAT
+
+#define DT_DRV_COMPAT  bosch_bmp390
+DT_INST_FOREACH_STATUS_OKAY_VARGS(BMP3XX_INST, BMP390_ID)
```

Comment:
clang-format needed here.

Reply:
Apply for some of them, but the
+	static const struct bmp388_config bmp388_config_##inst##chipid = {                         \
+		BMP388_BUS_CFG(inst) BMP388_INT_CFG(inst).iir_filter =                             \
+			DT_INST_ENUM_IDX(inst, iir_filter),         
is not valid. , .iir_filter should be on its own line.